server/src/main/java/keywhiz/service/daos/SecretDAO.java

```diff
@@ -198,6 +198,33 @@ public long createOrUpdateSecret(String name, String encryptedSecret, String cre
     return ImmutableList.copyOf(results);
   }
 
+  /**
+   * @param name of secret series to look up secrets by.
+   * @param newestIdx the first index to select in a list of versions sorted by creation time
+   * @param oldestIdx the last index to select in a list of versions sorted by creation time
+   * @return Versions of a secret matching input parameters or Optional.absent().
+   */
+  public Optional<SecretSeriesAndVersions> getSecretVersionsByName(String name, int newestIdx,
+      int oldestIdx) {
+    checkArgument(!name.isEmpty());
+    checkArgument(newestIdx >= 0);
+    checkArgument(oldestIdx >= 0);
+
+    SecretContentDAO secretContentDAO = secretContentDAOFactory.using(dslContext.configuration());
+    SecretSeriesDAO secretSeriesDAO = secretSeriesDAOFactory.using(dslContext.configuration());
+
+    Optional<SecretSeries> series = secretSeriesDAO.getSecretSeriesByName(name);
+    if (series.isPresent()) {
+      long secretId = series.get().id();
+      Optional<ImmutableList<SecretContent>> contents =
+          secretContentDAO.getSecretVersionsBySecretId(secretId, newestIdx, oldestIdx);
```

Comment:
Could this be made simpler/more efficient by using a join to get the secret versions by the name in one query, rather than using the name to get an id and using the id to get the versions? It's only two tables, with a primary/foreign key link, so I don't think a join would be over-complicated.